### PR TITLE
Some PartyBots tweaks

### PR DESCRIPTION
**Pull request by GrenderG**
## 🍰 Pullrequest
- Minor code cleanups.
- Split Bot / PlayerBot / PartyBot config so it's more understandable.
- Added config to set a maximum of PartyBots (although it can be easily skipped if you spam the command fast enough).
- Added config to skip PartyBots spawning restriction checks.
- PartyBots added with .partybot add will now have the same level as the summoner.

### CODE REVIEW

**Comment by iProfessorZ** [comment]
Nice work! Have you considered porting the ike3-bot code?:D

**Comment by GrenderG** [comment]
> Nice work! Have you considered porting the ike3-bot code?:D

I never looked into those bots, not sure how hard would be to port them.

**Comment by ratkosrb** [comment]
> Have you considered porting the ike3-bot code?:D

I do not approve of that. This is not a singleplayer project.

File: src/mangosd/mangosd.conf.dist.in [hunk 1]
```diff
@@ -2812,16 +2812,20 @@ OfflineChat.IP     = 0.0.0.0
 OfflineChat.Password= "p8zqkt"
 
 ###################################################################################################################
-#    Player bots
+#    Bots
```
Comment (ratkosrb) [inline src/mangosd/mangosd.conf.dist.in]: why did you remove the word player, bring it back

File: src/game/World.cpp [hunk 2]
```diff
@@ -911,7 +911,9 @@ void World::LoadConfigSettings(bool reload)
     setConfigMinMax(CONFIG_UINT32_RELOCATION_VMAP_CHECK_TIMER, "Movement.RelocationVmapsCheckDelay", 0, 0, 2000);
 
     sPlayerBotMgr.LoadConfig();
-    setConfig(CONFIG_BOOL_PLAYER_BOT_SHOW_IN_WHO_LIST, "PlayerBot.ShowInWhoList", false);
+    setConfig(CONFIG_BOOL_BOT_SHOW_IN_WHO_LIST, "Bot.ShowInWhoList", false);
```
Comment (ratkosrb) [inline src/game/World.cpp]: change it back to PlayerBot

File: src/mangosd/mangosd.conf.dist.in (hunk 1 above)
Comment (GrenderG) [reply src/mangosd/mangosd.conf.dist.in]: Because a Bot can be either a PlayerBot or a PartyBot, this way the configuration is easily understandable:

- `Bot`: Configuration affecting both PlayerBot and PartyBot.
- `PlayerBot`: Only affects PlayerBot.
- `PartyBot`: Only affects PartyBot.

(See how `Bot.ShowInWhoList` affects both of them, for example).
